src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -1081,6 +1081,128 @@ static Value builtinStringRaw(ExecutionState& state, Value thisValue, size_t arg
     }
 }
 
+// https://www.ecma-international.org/ecma-262/8.0/#sec-string.prototype.padstart
+// 21.1.3.14String.prototype.padStart( maxLength [ , fillString ] )
+static Value builtinStringPadStart(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    // Let O be ? RequireObjectCoercible(this value).
+    // Let S be ? ToString(O).
+    RESOLVE_THIS_BINDING_TO_STRING(S, String, padStart);
+
+    // Let intMaxLength be ? ToLength(maxLength).
+    // Let stringLength be the number of elements in S.
+    size_t intMaxLength = 0;
+    if (argc >= 1) {
+        intMaxLength = argv[0].toInteger(state);
+    }
+    size_t stringLength = S->length();
+
+    // If intMaxLength is not greater than stringLength, return S.
+    if (intMaxLength <= stringLength) {
+        return S;
+    }
+
+    // If fillString is undefined, let filler be a String consisting solely of the code unit 0x0020 (SPACE).
+    // Else, let filler be ? ToString(fillString).
+    String* filler;
+    if (argc >= 2) {
+        filler = argv[1].toString(state);
+    } else {
+        filler = String::fromCharCode(0x0020);
```

Comment:
Could you use StaticStrings.asciiTable[0x20].string() instead of creating newone?

Reply:
I was not aware of this possibility, thanks for bringing it up!